src/cl/basic/cl_basic_context.c

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "cl_basic.h"
+#include "cl/ucc_cl_log.h"
+#include "utils/ucc_malloc.h"
+
+UCC_CLASS_INIT_FUNC(ucc_cl_basic_context_t,
+                    const ucc_base_context_params_t *params,
+                    const ucc_base_config_t *config)
+{
+    ucc_cl_context_config_t *cl_config =
+        ucc_derived_of(config, ucc_cl_context_config_t);
```

Comment:
Does compiler allow to discard const for config? Looks like ucc_cl_context_config_t  should be const pointer

Reply:
fixed (compiler allowed probably due to derived_of casting, it is strange though)